### PR TITLE
Added the challenge objectives - recursive sorts

### PR DESCRIPTION
**Pull request by Estrax**

### CODE REVIEW

File: src/recursive_sorting/recursive_sorting.py [hunk 1]
```diff
@@ -1,33 +1,121 @@
...
+        arr = merge_in_place(arr, l, m, r)
     return arr
 
 
 # STRETCH: implement the Timsort function below
 # hint: check out https://github.com/python/cpython/blob/master/Objects/listsort.txt
-def timsort( arr ):
+def timsort(arr):
+    if len(arr) < 64:
+        for i in range(len(arr)):
```
Comment (Estrax) [inline src/recursive_sorting/recursive_sorting.py:79]: could be a separate function for insertion sort

File: src/recursive_sorting/recursive_sorting.py [hunk 2]
```diff
@@ -1,33 +1,121 @@
...
+                if not new_run:
+                    runs.append([arr[i]])
+                    new_run.append(arr[i])
+                else:
+                    runs.append(new_run)
+                    new_run = []
+            else:
+                new_run.append(arr[i])
+        for run in runs:
+            for i in range(len(run)):
```
Comment (Estrax) [inline src/recursive_sorting/recursive_sorting.py:107]: as there is also an insertion sort

File: src/recursive_sorting/recursive_sorting.py [hunk 3]
```diff
@@ -1,33 +1,121 @@
...
+                arr[j+1] = arr[j]
+                j -= 1
+            arr[j+1] = key
+    else:
+        runs = []
+        sorted_runs = []
+        arr_len = len(arr)
+        new_run = [arr[0]]
+
+        for i in range(1, arr_len):
```
Comment (Estrax) [inline src/recursive_sorting/recursive_sorting.py:92]: can be done better with up to log n memory at most cases

Comment (Estrax) [reply src/recursive_sorting/recursive_sorting.py:92]: I just need to spend more time with it on paper/whiteboard.

File: src/recursive_sorting/recursive_sorting.py [hunk 4]
```diff
@@ -1,33 +1,121 @@
...
+        i += 1
+    while j < len(arrA):
+        merged_arr[i] = arrA[j]
+        j += 1
+        i += 1
+    while k < len(arrB):
+        merged_arr[i] = arrB[k]
+        k += 1
+        i += 1
     return merged_arr
```
Comment (StephenBondor) [inline src/recursive_sorting/recursive_sorting.py:25]: Excellent! See if you can refactor merged_arr down to around 15 lines instead of 25ish

**Review by StephenBondor — APPROVED** [review]
great work!

File: src/recursive_sorting/recursive_sorting.py (hunk 4 above)
Comment (StephenBondor) [reply src/recursive_sorting/recursive_sorting.py:25]: Whoops! I meant the merge() function

**Review by StephenBondor — COMMENTED** [review]
corrected a comment

**Pull request merged**